test/integration/build/test_build.py

```diff
@@ -81,6 +83,97 @@ def test_app_without_rockspec(cartridge_cmd, project_without_dependencies):
     assert 'Application directory should contain rockspec' in output
 
 
+def test_app_with_rockspec_bad_name(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    bad_name_rockspec = "bad_rockspec-scm-1.rockspec"
+    rocks_make_output = "Rockspec %s doesn't exist" % (bad_name_rockspec)
+
+    # with --spec
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        bad_name_rockspec,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 1, 'Building project should fail'
+    assert rocks_make_output in output
+
+
+def test_app_with_rockspec_from_other_dir(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    dir_name = 'some_dir'
+    dir_path = os.path.join(project.path, dir_name)
+    os.mkdir(dir_path)
+
+    version = 'scm-2'
+    rockspec_name = '%s-%s.rockspec' % (project.name, version)
+    rockspec_path = os.path.join(dir_path, rockspec_name)
+    who_am_i = 'I am %s' % rockspec_name
+
+    set_whoami_build_specs(dir_path, project.name, version, rockspec_name)
+
+    build_log = 'Running `tarantoolctl rocks make %s`' % rockspec_path
```

Comment:
Let's move `build_log` declaration closer to the check. It improve code structure - we have several blocks that performs separated checks.

```python
cmd = [...]

# here we run command and check it's rc
rc, output = run_command_and_get_output(cmd, cwd=project.path)
assert rc == 0

# here we check command output
build_log = ...
assert build_log in output
assert who_am_i in output
```

Comments in snippet above are for you, don't add them to patch.